fileUpload/fileUpload.go

```diff
@@ -0,0 +1,151 @@
+package fileupload
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/url"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	pbDataSync "go.viam.com/api/app/datasync/v1"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/utils/rpc"
+)
+
+const (
+	UploadChunkSize = 32 * 1024 // size of the data included in each message of a FileUpload stream.
+	imageName       = "preview.jpeg"
+	appURL          = "https://app.viam.com:443"
+)
+
+// uploadJpeg uploads a jpeg thumbnail of a pointcloud used for a slam map package.
+func UploadJpeg(
+	ctx context.Context,
+	content *bytes.Buffer,
+	partID string, // you must have the partID for the robot
+	apiKey, apiKeyID string, // i think the robot api key you already use should work, but if it doesn't we can easily up the permissions on that key
+	logger logging.Logger,
+) (string, error) {
+	syncClient, conn, err := connectToApp(ctx, apiKey, apiKeyID, logger)
+	defer conn.Close()
+
+	stream, err := syncClient.FileUpload(ctx)
+	if err != nil {
+		return "", err
+	}
+	md := &pbDataSync.UploadMetadata{
+		// NOTE: Passing the PartID is temp.
+		// Once we move to use Org Keys for authenticating with App
+		// the PartID field can be removed in favor of sending the
+		// OrgKey
+		PartId:        partID,
+		Type:          pbDataSync.DataType_DATA_TYPE_FILE, // i used this in the past and saw my images show up in data, but it is possible this is no longer correct
+		FileName:      filepath.Base(imageName),
+		FileExtension: filepath.Ext(imageName),
+		Tags: []string{
+			"TAG1", // generic for all canary uploads
+			"TAG2", // specific for this run
+		},
```

Comment:
think we want 3 tags, something like

```
"ROVER_CANARY_TESTS", // generic for all canary uploads
"<test date/timestamp>", // specific for this run
"<MOTOR/SETVEL/GRID>", // specific for this type of image(grid vs motor)
```

for the date/timestamp we can prob just use the date, and not worry if multiple runs have the same tag for if we have to do some manual runs